core/lib/Thelia/Model/Category.php

```diff
@@ -42,22 +44,22 @@ public function getRewrittenUrlViewName()
      *
      * /!\ the number of queries is exponential, use it with caution
      *
+     * @param bool|string $productVisibility: true (default) to count only visible products, false to count only hidden
+     *                    products, or * to count all products.
      * @return int
      */
-    public function countAllProducts()
+    public function countAllProducts($productVisibility = true)
     {
         $children = CategoryQuery::findAllChild($this->getId());
```

Comment:
I agree. Unfortunately, ProductQuery provides only a filterByCategory() method, not a filterByCategoryId().